tools/flye/flye.xml

```diff
@@ -0,0 +1,103 @@
+<tool id="flye" name="Assembly" version="0.1.0"
+    <description>of long and error-prone reads</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements" />
+    <version_command>flye --version</version_command>
+    <command detect_errors="exit_code">
+    <![CDATA[
+
+    #for $counter, $input in enumerate($inputs):
+        ln -s '$input' ./input_${counter}.${input.ext} &&
+    #end for
+    
+    flye
+    $mode
+    #for $counter, $input in enumerate($inputs):
+        ./input_${counter}.${input.ext}
+    #end for
+
+    -o out_dir 
+    -g '$g'
+    -t \${GALAXY_SLOTS:-4}
+    -i $i
+    #if $m:
+        -m '$m'
+    #end if    
+    2>&1
+    ]]></command>
+    <inputs>
+        <param name="inputs" type="data" format="fasta,fasta.gz,fastq,fastq.gz" multiple="true" label="Input reads" />
+        <param name="mode" type="select" label="Mode">
+            <option value="--nano-raw">Nanopore raw</option>
+            <option value="--nano-corr">Nanopore corrected</option>
+            <option value="--pacbio-raw">PacBio raw</option>
+            <option value="--pacbio-corr">PacBio corrected</option>
+            <option value="--subassemblies">high-quality contig-like input</option>
+        </param>
+        <param argument="-g" type="text" label="estimated genome size (for example, 5m or 2.6g)" />
```

Comment:
If we find the estimated genome sizes ... we could add something like:

```
    <xml name="effectiveGenomeSize">
        <conditional name="effectiveGenomeSize">
            <param name="effectiveGenomeSize_opt" type="select" label="Effective genome size"
                help="The effective genome size is the portion of the genome that is mappable. Large fractions of the genome are stretches of NNNN that should be discarded.
                    Also, if repetitive regions were not included in the mapping of reads, the effective genome size needs to be adjusted accordingly.
                    We provide a table of useful sizes here: http://deeptools.readthedocs.io/en/latest/content/feature/effectiveGenomeSize.html">
                <option value="93260000">ce10 (93260000)</option>
                <option value="130428560">dm3 (130428560)</option>
                <option value="125464728">dm6 (125464728)</option>
                <option value="2685511504" selected="true">hg19/GRCh37 (2685511504)</option>
                <option value="2701495761">GRCh38/hg38 (2701495761)</option>
                <option value="2304947926">GRCm37/mm9 (2304947926)</option>
                <option value="2308125349">GRCm38/mm10 (2308125349)</option>
                <option value="specific">user specified</option>
            </param>
            <when value="specific">
                <param argument="--effectiveGenomeSize" type="integer" value="" label="Effective genome size"
                       help="e.g. ce10: 93260000, dm3: 130428560, hg19: 2685511504, mm9: 2304947926"/>
            </when>
            <when value="93260000" />
            <when value="130428560" />
            <when value="125464728" />
            <when value="2685511504" />
            <when value="2701495761" />
            <when value="2304947926" />
            <when value="2308125349" />
        </conditional>
</xml>
```

Reply:
It would be a convenient asset to help the user. Still I would prefer to not guide the user, since flye deals with repeated regions, the sizes likely differ from short read 'effective genome sizes'..

Reply:
flye wants to have the `genome size` isn't it? This pasted one was just an example, we should __not__ use the effective genome size, but we could search for the `genome size` of the most common genomes and offer something similar.

Reply:
As a minimum we should add a validator to the text parameter I think.

Reply:
Could u pls link to a  tool with the validator parser?

Reply:
May we skip this for this PR? :) If input be wrong, the tool complains and fails anyway

Reply:
https://github.com/bgruening/galaxytools/blob/5c6486dead878a8c9521e1d6d50b3a537a2ec2b0/tools/trim_galore/trim_galore.xml#L20
and
https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-inputs-param-validator

Reply:
added